.github/workflows/pr-build.yaml

```diff
@@ -149,9 +148,45 @@ jobs:
           PR_NUMBER: ${{ github.event.pull_request.number }}
         run: |
           . ./hack/ci/setup-env.sh
-      - run: make build-app-image-${{ matrix.APP }}
+      - run: |
+          make build-app-image-${{ matrix.APP }}
+          make save-app-image-${{ matrix.APP }}
+      - name: Upload artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.APP }}-${{github.sha}}
+          path: /tmp/${{ matrix.APP }}.tar
+          retention-days: 1
+
+  push-apps:
```

Comment:
I thought about sharing a common part, e.g.
```yaml
  push: &genericmetapush
    runs-on: ubuntu-latest
    if: github.event.pull_request.draft == false
    permissions:
      contents: read
      packages: write
  steps: &genericsteps
    - name: Checkout code
      uses: actions/checkout@v2
      with:
        persist-credentials: false
    - name: Download artifact
      uses: actions/download-artifact@v2
      with:
        name: ${{ matrix.APP }}-${{github.sha}}
        path: /tmp
    - name: Log into registry
      run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
    - name: Setup environment
      env:
        PR_NUMBER: ${{ github.event.pull_request.number }}
      run: |
        . ./hack/ci/setup-env.sh

  push-apps: *genericmetapush
    name: Push ${{ matrix.APP }}
    needs: [ prepare-matrix, build-apps ]
    strategy:
      matrix: ${{fromJson(needs.prepare-matrix.outputs.matrix-app)}}
    steps: *genericsteps
    - name: Load image
      run: docker load --input /tmp/${{ matrix.APP }}.tar
    - run: make push-app-image-${{ matrix.APP }}
```

but after trying that I see that it decreases readability and at the same time may cause some security issues.